server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java

```diff
@@ -182,6 +184,75 @@ public void getMessageListShouldListMessageWhenTheUserHasOnlyReadRight() throws
             .body(ARGUMENTS + ".messageIds", contains(message.getMessageId().serialize()));
     }
 
+    @Test
+    public void getMessageListShouldNotFilterMessagesWhenTextFilterMatchesBodyAfterTheMessageMailboxHasBeenChanged() throws Exception {
```

Comment:
The idea here is that before the bugfix setMessage has the effect to reindex the message in ES but only with the metadata we did loose everything related to the content on the email (from, subject, body, etc.., etc..)